src/main/java/org/humancellatlas/ingest/security/common/jwk/JwtVerifierResolver.java

```diff
@@ -0,0 +1,43 @@
+package org.humancellatlas.ingest.security.common.jwk;
+
+import com.auth0.jwt.JWT;
+import com.auth0.jwt.algorithms.Algorithm;
+import com.auth0.jwt.interfaces.DecodedJWT;
+import com.auth0.jwt.interfaces.JWTVerifier;
+
+import java.security.interfaces.RSAPublicKey;
+import java.util.Optional;
+
+/**
+ * Helper class whose main purpose is to create a {@link JWTVerifier} instance given a JWT string.
+ *
+ * This is part of the wrapper subsystem to help compartmentalise the area of the application that relies on Auth0's
+ * library for processing JWTs.
+ */
+public class JwtVerifierResolver {
+
+    private final JwkVault jwkVault;
+
+    private final Optional<String> audience;
+    private final Optional<String> issuer;
+
+    public JwtVerifierResolver(JwkVault jwkVault, String audience, String issuer) {
+        this.jwkVault = jwkVault;
+        this.audience = Optional.ofNullable(audience);
+        this.issuer = Optional.ofNullable(issuer);
+    }
+
+    public String getIssuer() {
+        return issuer.get();
```

Comment:
Is it best to just return the Optional issuer here? .get() will throw an exception if issue isn't present.

Reply:
The Optional is used here mostly to sanitise internal implementation. The getter remains pretty much nullable as it was before.